homeassistant/components/sonos/number.py

```diff
@@ -0,0 +1,64 @@
+"""Entity representing a Sonos number control."""
+from __future__ import annotations
+
+from homeassistant.components.number import NumberEntity
+from homeassistant.const import ENTITY_CATEGORY_CONFIG
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+
+from .const import SONOS_CREATE_LEVELS
+from .entity import SonosEntity
+from .helpers import soco_error
+from .speaker import SonosSpeaker
+
+LEVEL_TYPES = ("bass", "treble")
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Sonos number platform from a config entry."""
+
+    async def _async_create_entities(speaker: SonosSpeaker) -> None:
```

Comment:
This doesn't need to be a coroutine function since we don't await inside.

Reply:
Thanks. Done in #60643.